ROADMAP.md

```diff
@@ -0,0 +1,154 @@
+## Vision
+
+> Amethyst aims to be a modular, parallel and data-driven game engine
+
+**Modularity** The game engine should be easily extendable to fit the needs of the game developers for their game idea. Therefore we expose an entity-component system, which allows to define new components and processors without needing to change the game engine code. This should emphasize the 'game engine as library' aspect.
+
+**Parallelism** Current and future CPUs are multi-core processors, which can run multiple tasks in parallel. Rust provides us useful language features to implement parallelism, which should be used to provide an easy and safe API for the engine.
+
+**Data-Driven** In order to achieve high performance, we will also have to take care of data-orientation next to parallelism. This is closely coupled with the ECS system: processors will perform certain operations over large data sets.
+
+## Goals
+These goals are mainly based on priority and will run through several iterations over the whole development process. So, short-term goals won't be finished when starting with mid-term content but still be worked on.
+### Short term
+These are the core functionalities for the game engine required to build a game. Mainly focuses on using the engine as library to build upon.
+* Entity-Component System
+* Rendering
+* Input Handling
+* Audio
+* Serialization/Reflection
+* Streaming/Asset Handling
+* Documentation/Tutorial
+
+### Mid-term
+
+* Physics
+* Tooling
+* Animation
+
+### Long-term
+
+* Scripting
+* Graphical UI
+* Networking
+* AI
+
+### Develop A Game
+A game engine is built in order to build games! The best tutorial for potential users are working examples, and fancier examples are even better.
+Under this assumption we should also try to spend some time on translating our implementations into a working game, showcasing the technical aspects of our engine.
+
+## Systems/Aspects
+Discussing the scopes and features of the different subsystems in more detail.
+
+### Entity-Component System
+Our current system is based on `specs` and should probably stick to it in near future (in case some magically new library pops up..).
+As the ECS is the core part of the architecture, this needs be carefully designed and should be clear under our 3 main vision aspects:
+
+**Modularity** We expose the API from `specs` to the developers, in order to allow them to implement new components, processors and resources (larger singleton objects). The engine will have built-in processors for common operations like `Transform` to ensure spatial hierarchy.
```

Comment:
I believe this is what the shard system (#34) is supposed to make possible. Should we also make that into a pr?